JSQMessagesViewController/Views/JSQMessagesCellTextView.m

```diff
@@ -42,6 +42,20 @@ - (void)awakeFromNib
                                  NSUnderlineStyleAttributeName : @(NSUnderlineStyleSingle | NSUnderlinePatternSolid) };
 }
 
+- (BOOL)pointInside:(CGPoint)point withEvent:(UIEvent *)event
+{
+    //Enable link selection and nothing else. https://stackoverflow.com/a/44878203
+    //Gesture recognizers will pick up everything else
+    UITextPosition *position = [self closestPositionToPoint:point];
+    if (position == nil) { return false; }
+    UITextRange *range = [self.tokenizer rangeEnclosingPosition:position withGranularity:UITextGranularityCharacter inDirection:UITextLayoutDirectionLeft];
```

Comment:
Will this work in right-to-left languages?

Reply:
I was actually wondering the same thing as I was working on this, but we are using the API to find the attribute for the tapped text and if that is a link. so selection direction is not a concern here.

Reply:
Makes sense re: attribute at character index